podcasts/SuggestedFoldersView.swift

```diff
@@ -1,16 +1,49 @@
 import PocketCastsDataModel
 import SwiftUI
+import PocketCastsServer
 
 class SuggestedFoldersModel: ObservableObject {
 
-    private static let somePodcastsUUIDs = ["e7abe050-6cc7-0130-f8c5-723c91aeae46", "ba993300-d71c-0137-1e26-0acc26574db2", "4eb5b260-c933-0134-10da-25324e2a541d", "71a77ab0-c8bf-0136-7b94-27f978dac4db", "467b49a0-c657-0138-e72e-0acc26574db2"]
+    @Published var folders: [SuggestedFolder] = []
 
-    var folders: [SuggestedFolder] = {
-        let result: [SuggestedFolder] = (0..<20).map { index in
-            SuggestedFolder(name: "Folder-\(index)", color: Int32(index), topPodcastUuids: somePodcastsUUIDs.shuffled())
+    enum State {
+        case start
+        case loading
+        case loaded
+        case failed
+    }
+
+    @Published var loadingState: State = .start
+
+    var failedToLoadAction: (() -> ())? = nil
+
+    init(failedToLoadAction: (() -> ())? = nil) {
+        self.failedToLoadAction = failedToLoadAction
+    }
+
+    func load() async {
+        if loadingState != .start {
+            return
         }
-        return result
-    }()
+        Task { @MainActor in
+            loadingState = .loading
```

Comment:
Perhaps `loadingState = .loading` should be on L27 (above the `Task`).

Otherwise two `load()` invocations, happening at the exact same time, might go trough?

Reply:
Fixed here: [217b4e7](https://github.com/Automattic/pocket-casts-ios/pull/2783/commits/217b4e7c98ed73e2ea935967edca60781f3eb359)